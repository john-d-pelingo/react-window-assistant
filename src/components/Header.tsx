import { AppBar, Link as MaterialLink, Toolbar } from '@material-ui/core'
import { grey, red } from '@material-ui/core/colors'
import { Link, LinkGetProps } from '@reach/router'
import React, { FC } from 'react'
import styled from 'styled-components'

import {
  colorClarifier,
  home,
  jsonSorter,
  keyCodeRevealer,
  textConverter,
  urlInterpreter,
} from 'consants/routes'
import { appName } from 'consants/strings'

const StyledToolbar = styled(Toolbar)`
  color: white;
  flex-wrap: wrap;
  justify-content: space-between;
`

const StyledLink = styled(Link)`
  && {
    margin: 8px 12px;
  }
`

const appRoutes: Array<{
  name: string
  path: string
}> = [
  { name: 'Home', path: home },
  { name: 'JSON Sorter', path: jsonSorter },
  { name: 'URL Interpreter', path: urlInterpreter },
  { name: 'Color Clarifier', path: colorClarifier },
  { name: 'Text Converter', path: textConverter },
  { name: 'Key Code Revealer', path: keyCodeRevealer },
]

const getActiveStyles = ({ isCurrent, href, location }: LinkGetProps) => {
  const isCompletelyCurrent = href === location.pathname || isCurrent
  return {
    style: {
      color: isCompletelyCurrent ? red['900'] : 'inherit',
      fontWeight: isCompletelyCurrent ? 'bold' : 'current',
    },
  }
}

export const Header: FC = () => {
  // TODO: transform to burger menu or something
  return (
    <AppBar
      color="default"
      elevation={0}
      position="static"
      style={{
        backgroundColor: grey['900'],
      }}
    >
      <StyledToolbar>
        <MaterialLink
          color="inherit"
          component={Link}
          noWrap
          to={home}
          variant="h6"
        >
          {appName}
        </MaterialLink>
        <nav>
          {appRoutes.map(appRoute => (
            <MaterialLink
              key={appRoute.path}
              color="textPrimary"
              component={StyledLink}
              getProps={getActiveStyles}
              to={appRoute.path}
            >
              {appRoute.name}
            </MaterialLink>
          ))}
        </nav>
      </StyledToolbar>
    </AppBar>
  )
}
