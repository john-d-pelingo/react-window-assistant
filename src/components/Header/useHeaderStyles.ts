import { createStyles, makeStyles } from '@material-ui/core'

export const useHeaderStyles = makeStyles(theme =>
  createStyles({
    appBar: {
      borderBottom: `1px solid ${theme.palette.divider}`,
    },
    toolbar: {
      flexWrap: 'wrap',
    },
    toolbarTitle: {
      flexGrow: 1,
    },
    link: {
      margin: theme.spacing(1, 1.5),
    },
  }),
)
