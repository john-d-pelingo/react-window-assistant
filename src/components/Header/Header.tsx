import { AppBar, Link as MaterialLink, Toolbar } from '@material-ui/core'
import { red } from '@material-ui/core/colors'
import { Link, LinkGetProps } from '@reach/router'
import React, { FC } from 'react'

import {
  colorClarifier,
  home,
  jsonSorter,
  urlInterpreter,
} from 'consants/routes'
import { appName } from 'consants/strings'

import { useHeaderStyles } from './useHeaderStyles'

const getActiveStyles = ({ isCurrent, href, location }: LinkGetProps) => {
  const isCompletelyCurrent = href === location.pathname || isCurrent
  return {
    style: {
      color: isCompletelyCurrent ? red['900'] : 'inherit',
      fontWeight: isCompletelyCurrent ? 'bold' : 'current',
    },
  }
}

export const Header: FC = () => {
  const classes = useHeaderStyles()

  // TODO: transform to burger menu or something
  return (
    <AppBar
      className={classes.appBar}
      color="default"
      elevation={0}
      position="static"
    >
      <Toolbar className={classes.toolbar}>
        <MaterialLink
          color="inherit"
          component={Link}
          noWrap
          to={home}
          variant="h6"
        >
          {appName}
        </MaterialLink>
        <nav>
          <MaterialLink
            className={classes.link}
            color="textPrimary"
            component={Link}
            getProps={getActiveStyles}
            to={home}
            variant="button"
          >
            Home
          </MaterialLink>
          <MaterialLink
            className={classes.link}
            color="textPrimary"
            component={Link}
            getProps={getActiveStyles}
            to={jsonSorter}
            variant="button"
          >
            JSON Sorter
          </MaterialLink>
          <MaterialLink
            className={classes.link}
            color="textPrimary"
            component={Link}
            getProps={getActiveStyles}
            to={urlInterpreter}
            variant="button"
          >
            URL Interpreter
          </MaterialLink>
          <MaterialLink
            className={classes.link}
            color="textPrimary"
            component={Link}
            getProps={getActiveStyles}
            to={colorClarifier}
            variant="button"
          >
            Color Clarifier
          </MaterialLink>
        </nav>
      </Toolbar>
    </AppBar>
  )
}
