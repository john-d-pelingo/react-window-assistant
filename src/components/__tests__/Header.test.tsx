import 'jest-styled-components'

import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'

import { Header } from '../Header'

let memoryHistory = createMemoryHistory()

describe('components - Header', () => {
  beforeEach(() => {
    memoryHistory = createMemoryHistory()
  })

  it('mounts', () => {
    expect(memoryHistory.location.pathname).toBe('/')

    const { container } = render(
      <Router history={memoryHistory}>
        <Header />
      </Router>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  // TODO: add test for each active link by using aria-label or text
})
