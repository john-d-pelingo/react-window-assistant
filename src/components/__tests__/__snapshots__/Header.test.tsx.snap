// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Header mounts 1`] = `
.c0 {
  color: white;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1.c1 {
  color: white;
  margin: 8px 12px;
}

<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault MuiPaper-elevation0"
  style="background-color: rgb(33, 33, 33);"
>
  <div
    class="MuiToolbar-root MuiToolbar-regular c0 MuiToolbar-gutters"
  >
    <a
      aria-current="page"
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap active"
      href="/"
    >
      Window Assistant
    </a>
    <nav>
      <a
        aria-current="page"
        class="c1 MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorTextPrimary active"
        href="/"
        style="color: rgb(183, 28, 28); font-weight: bold;"
      >
        Home
      </a>
      <a
        class="c1 MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorTextPrimary"
        href="/jsonsorter"
      >
        JSON Sorter
      </a>
      <a
        class="c1 MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorTextPrimary"
        href="/urlinterpreter"
      >
        URL Interpreter
      </a>
      <a
        class="c1 MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorTextPrimary"
        href="/colorclarifier"
      >
        Color Clarifier
      </a>
      <a
        class="c1 MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorTextPrimary"
        href="/textconverter"
      >
        Text Converter
      </a>
      <a
        class="c1 MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorTextPrimary"
        href="/keycoderevealer"
      >
        Key Code Revealer
      </a>
    </nav>
  </div>
</header>
`;
